src/App.tsx

```diff
@@ -3,33 +3,56 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    this.setState({ query: value.toLowerCase() });
+  };
 
   render() {
+    const { query } = this.state;
+
+    const visibleMovies = moviesFromServer.filter(movie => (
+      movie.description.toLowerCase().includes(query)
+      || movie.title.toLowerCase().includes(query)
+    ));
```

Comment:
Take this logic to a separate method and call it in the `render` method. For example: `getVisibleMovies()`